New Thrift, new errors

[Via Go kit](https://travis-ci.org/go-kit/kit/jobs/352690700#L700).

```
# github.com/openzipkin/zipkin-go-opentracing/thrift/gen-go/scribe
../../openzipkin/zipkin-go-opentracing/thrift/gen-go/scribe/scribe.go:282:14: not enough arguments in call to oprot.Flush
	have ()
	want (context.Context)
../../openzipkin/zipkin-go-opentracing/thrift/gen-go/scribe/scribe.go:299:16: not enough arguments in call to oprot.Flush
	have ()
	want (context.Context)
../../openzipkin/zipkin-go-opentracing/thrift/gen-go/scribe/scribe.go:312:16: not enough arguments in call to oprot.Flush
	have ()
	want (context.Context)
../../openzipkin/zipkin-go-opentracing/thrift/gen-go/scribe/scribe.go:326:24: not enough arguments in call to oprot.Flush
	have ()
	want (context.Context)
```

--- a/collector-kafka_test.go
+++ b/collector-kafka_test.go
@@ -1,6 +1,7 @@
 package zipkintracer
 
 import (
+	"context"
 	"errors"
 	"testing"
 	"time"
@@ -167,7 +168,7 @@ func deserializeSpan(t *testing.T, e sarama.Encoder) *zipkincore.Span {
 	s := zipkincore.NewSpan()
 	mb := thrift.NewTMemoryBufferLen(len(bytes))
 	_, _ = mb.Write(bytes)
-	_ = mb.Flush()
+	_ = mb.Flush(context.Background())
 	pt := thrift.NewTBinaryProtocolTransport(mb)
 	err = s.Read(pt)
 	if err != nil {

--- a/thrift/gen-go/scribe/scribe-consts.go
+++ b/thrift/gen-go/scribe/scribe-consts.go
@@ -5,8 +5,8 @@ package scribe
 
 import (
 	"bytes"
-	"reflect"
 	"context"
+	"reflect"
 	"fmt"
 	"github.com/apache/thrift/lib/go/thrift"
 )

--- a/thrift/gen-go/scribe/scribe-remote/scribe-remote.go
+++ b/thrift/gen-go/scribe/scribe-remote/scribe-remote.go
@@ -27,6 +27,22 @@ func Usage() {
   os.Exit(0)
 }
 
+type httpHeaders map[string]string
+
+func (h httpHeaders) String() string {
+  var m map[string]string = h
+  return fmt.Sprintf("%s", m)
+}
+
+func (h httpHeaders) Set(value string) error {
+  parts := strings.Split(value, ": ")
+  if len(parts) != 2 {
+    return fmt.Errorf("header should be of format 'Key: Value'")
+  }
+  h[parts[0]] = parts[1]
+  return nil
+}
+
 func main() {
   flag.Usage = Usage
   var host string
@@ -35,7 +51,8 @@ func main() {
   var urlString string
   var framed bool
   var useHttp bool
-  var parsedUrl url.URL
+  headers := make(httpHeaders)
+  var parsedUrl *url.URL
   var trans thrift.TTransport
   _ = strconv.Atoi
   _ = math.Abs
@@ -46,16 +63,18 @@ func main() {
   flag.StringVar(&urlString, "u", "", "Specify the url")
   flag.BoolVar(&framed, "framed", false, "Use framed transport")
   flag.BoolVar(&useHttp, "http", false, "Use http")
+  flag.Var(headers, "H", "Headers to set on the http(s) request (e.g. -H \"Key: Value\")")
   flag.Parse()
   
   if len(urlString) > 0 {
-    parsedUrl, err := url.Parse(urlString)
+    var err error
+    parsedUrl, err = url.Parse(urlString)
     if err != nil {
       fmt.Fprintln(os.Stderr, "Error parsing URL: ", err)
       flag.Usage()
     }
     host = parsedUrl.Host
-    useHttp = len(parsedUrl.Scheme) <= 0 || parsedUrl.Scheme == "http"
+    useHttp = len(parsedUrl.Scheme) <= 0 || parsedUrl.Scheme == "http" || parsedUrl.Scheme == "https"
   } else if useHttp {
     _, err := url.Parse(fmt.Sprint("http://", host, ":", port))
     if err != nil {
@@ -68,6 +87,12 @@ func main() {
   var err error
   if useHttp {
     trans, err = thrift.NewTHttpClient(parsedUrl.String())
+    if len(headers) > 0 {
+      httptrans := trans.(*thrift.THttpClient)
+      for key, value := range headers {
+        httptrans.SetHeader(key, value)
+      }
+    }
   } else {
     portStr := fmt.Sprint(port)
     if strings.Contains(host, ":") {
@@ -132,7 +157,7 @@ func main() {
       Usage()
       return
     }
-    factory8 := thrift.NewTSimpleJSONProtocolFactory()
+    factory8 := thrift.NewTJSONProtocolFactory()
     jsProt9 := factory8.GetProtocol(mbTrans6)
     containerStruct0 := scribe.NewScribeLogArgs()
     err10 := containerStruct0.ReadField1(jsProt9)

--- a/thrift/gen-go/scribe/scribe.go
+++ b/thrift/gen-go/scribe/scribe.go
@@ -5,10 +5,10 @@ package scribe
 
 import (
 	"bytes"
+	"context"
 	"reflect"
 	"database/sql/driver"
 	"errors"
-	"context"
 	"fmt"
 	"github.com/apache/thrift/lib/go/thrift"
 )
@@ -210,14 +210,12 @@ type ScribeClient struct {
   c thrift.TClient
 }
 
-// Deprecated: Use NewScribe instead
 func NewScribeClientFactory(t thrift.TTransport, f thrift.TProtocolFactory) *ScribeClient {
   return &ScribeClient{
     c: thrift.NewTStandardClient(f.GetProtocol(t), f.GetProtocol(t)),
   }
 }
 
-// Deprecated: Use NewScribe instead
 func NewScribeClientProtocol(t thrift.TTransport, iprot thrift.TProtocol, oprot thrift.TProtocol) *ScribeClient {
   return &ScribeClient{
     c: thrift.NewTStandardClient(iprot, oprot),
@@ -230,13 +228,16 @@ func NewScribeClient(c thrift.TClient) *ScribeClient {
   }
 }
 
+func (p *ScribeClient) Client_() thrift.TClient {
+  return p.c
+}
 // Parameters:
 //  - Messages
 func (p *ScribeClient) Log(ctx context.Context, messages []*LogEntry) (r ResultCode, err error) {
   var _args0 ScribeLogArgs
   _args0.Messages = messages
   var _result1 ScribeLogResult
-  if err = p.c.Call(ctx, "Log", &_args0, &_result1); err != nil {
+  if err = p.Client_().Call(ctx, "Log", &_args0, &_result1); err != nil {
     return
   }
   return _result1.GetSuccess(), nil
@@ -279,7 +280,7 @@ func (p *ScribeProcessor) Process(ctx context.Context, iprot, oprot thrift.TProt
   oprot.WriteMessageBegin(name, thrift.EXCEPTION, seqId)
   x3.Write(oprot)
   oprot.WriteMessageEnd()
-  oprot.Flush()
+  oprot.Flush(ctx)
   return false, x3
 
 }
@@ -296,7 +297,7 @@ func (p *scribeProcessorLog) Process(ctx context.Context, seqId int32, iprot, op
     oprot.WriteMessageBegin("Log", thrift.EXCEPTION, seqId)
     x.Write(oprot)
     oprot.WriteMessageEnd()
-    oprot.Flush()
+    oprot.Flush(ctx)
     return false, err
   }
 
@@ -309,7 +310,7 @@ var retval ResultCode
     oprot.WriteMessageBegin("Log", thrift.EXCEPTION, seqId)
     x.Write(oprot)
     oprot.WriteMessageEnd()
-    oprot.Flush()
+    oprot.Flush(ctx)
     return true, err2
   } else {
     result.Success = &retval
@@ -323,7 +324,7 @@ var retval ResultCode
   if err2 = oprot.WriteMessageEnd(); err == nil && err2 != nil {
     err = err2
   }
-  if err2 = oprot.Flush(); err == nil && err2 != nil {
+  if err2 = oprot.Flush(ctx); err == nil && err2 != nil {
     err = err2
   }
   if err != nil {

--- a/thrift/gen-go/zipkincore/zipkinCore-consts.go
+++ b/thrift/gen-go/zipkincore/zipkinCore-consts.go
@@ -5,8 +5,8 @@ package zipkincore
 
 import (
 	"bytes"
-	"reflect"
 	"context"
+	"reflect"
 	"fmt"
 	"github.com/apache/thrift/lib/go/thrift"
 )

--- a/thrift/gen-go/zipkincore/zipkinCore.go
+++ b/thrift/gen-go/zipkincore/zipkinCore.go
@@ -5,10 +5,10 @@ package zipkincore
 
 import (
 	"bytes"
+	"context"
 	"reflect"
 	"database/sql/driver"
 	"errors"
-	"context"
 	"fmt"
 	"github.com/apache/thrift/lib/go/thrift"
 )

